browser/brave_shields/domain_block_page_browsertest.cc

```diff
@@ -331,9 +345,35 @@ IN_PROC_BROWSER_TEST_F(DomainBlockTest, NoThirdPartyInterstitial) {
   ASSERT_STRNE("chrome-error://chromewebdata/", location.c_str());
 }
 
+IN_PROC_BROWSER_TEST_F(DomainBlockTest, NoInterstitialUnlessAggressive) {
+  ASSERT_TRUE(InstallDefaultAdBlockExtension());
+  GURL url = embedded_test_server()->GetURL("a.com", "/simple.html");
+  SetAdControlType(content_settings(), ControlType::BLOCK, url);
+
+  // Navigate to a page on a.com. This should work normally.
+  NavigateTo(url);
+  ASSERT_FALSE(IsShowingInterstitial());
+
+  // Block a.com in rules but allow a.com via shields, then attempt to navigate
+  // to a page on a.com. This should not show an interstitial.
+  BlockDomainByURL(url);
+  SetAdControlType(content_settings(), ControlType::ALLOW, url);
```

Comment:
Might be helpful to have a test for "normal" ads/trackers blocking mode (i.e. `ControlType::BLOCK_THIRD_PARTY`)

Reply:
added in https://github.com/brave/brave-core/commit/7ea545515d48b6a3678cbd1244d568a312dc6826

Reply:
hmm, I still can't find it - am I missing something? :confused:

Reply:
https://github.com/brave/brave-core/blob/master/components/brave_shields/browser/brave_shields_util.cc#L156 so I'm calling https://github.com/brave/brave-core/blob/master/components/brave_shields/browser/brave_shields_util.cc#L127 instead.

Reply:
Gotcha! Looks good then.